Add Webtoons plugin

browningluke opened the pull request:
This should work for all Webtoons on the English site [Webtoons](https://www.webtoons.com/en/).

Let me know if you see any obvious bugs and what you think!

browningluke committed:
Add Webtoons plugin

hkalexling commented:
Thanks for the PR!

I was trying it and noticed a few issues. Maybe I am missing something?

- I tried to search "hello" and got the error "Search failed. Error: Hostname lookup for m.webtoons.comundefined failed: No address found". But searching for other random words like "brah" works as expected and gives "Search failed. Error: Could not find a webtoon with that title."
- I searched for "boyfriends", which should return [a matching manga](https://www.webtoons.com/en/slice-of-life/boyfriends/list?title_no=2616). But instead, I got the error "Search failed. Error: Failed to get a chapter ID."

browningluke committed:
Webtoons: Fix search & genre bug

- Fixed bug including authors in search.
- Added support for hyphens in genre ID (e.g. slice-of-life)

browningluke commented:
Sorry about that!

> - I tried to search "hello" and got the error "Search failed. Error: Hostname lookup for m.webtoons.comundefined failed: No address found". But searching for other random words like "brah" works as expected and gives "Search failed. Error: Could not find a webtoon with that title."

This was due to not excluding authors from the search response, which I have now done.

> - I searched for "boyfriends", which should return a matching manga. But instead, I got the error "Search failed. Error: Failed to get a chapter ID."

This was due to not supporting hyphens in the genre id (e.g. 'action' vs 'slice-of-life'). This is also fixed now.

It should now work properly.

hkalexling commented:
Works like a charm now. Thanks!
